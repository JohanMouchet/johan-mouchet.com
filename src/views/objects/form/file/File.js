// @flow

import * as React from "react";
import cx from "classnames";
import { IconCloudUpload } from "views/objects";
import "./File.scss";

type Props = {
  id?: string,
  name: string,
  multipleCaption?: string,
  className?: string | Array<string> | Object,
  attributes?: { [key: string]: any },
  children?: Node,
};

const File = ({
  id,
  name,
  multipleCaption,
  className,
  attributes,
  children,
}: Props) => {
  const classNames = cx("o-form__file", "o-button", className);

  return (
    <label className={classNames}>
      <input
        id={id}
        type="file"
        name={name}
        className="o-form__file-input"
        data-multiple-caption={
          multipleCaption ? multipleCaption : "{count} files selected"
        }
        {...attributes}
      />
      <span className="o-form__file-icon">
        <IconCloudUpload />
      </span>
      {children && <span className="o-form__file-text">{children}</span>}
    </label>
  );
};

export default File;
