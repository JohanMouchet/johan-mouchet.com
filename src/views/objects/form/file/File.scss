@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";
// @import "assets/styles/utilities/_utilities";

.o-form__file {
  &-input {
    // TODO: find a way to import single utility
    // @extend .u-is-visually-hidden;
  }

  &-icon {
    svg {
      font-size: t-modular-scale(1);
      vertical-align: middle;
    }
  }

  &-text {
    margin-left: 0.5em;
  }
}
