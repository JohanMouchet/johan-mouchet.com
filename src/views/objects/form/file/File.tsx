import React from "react";
import cx from "classnames";
import { IconCloudUpload } from "views/objects";
import buttonStyles from "../../button/Button.module.scss";
import styles from "./File.module.scss";

type Props = {
  id?: string;
  name: string;
  multipleCaption?: string;
  label?: string;
  className?: string | string[] | { [key: string]: boolean };
  attributes?: boolean | number | string;
  children?: React.ReactNode;
};

const File: React.FC<Props> = ({
  id,
  name,
  multipleCaption,
  label = "Upload",
  className,
  children,
  ...attributes
}) => {
  const classNames = cx(
    styles["o-form__file"],
    buttonStyles["o-button"],
    className
  );

  return (
    <label className={classNames}>
      <input
        id={id}
        type="file"
        name={name}
        className="o-form__file-input u-is-visually-hidden"
        data-multiple-caption={
          multipleCaption ? multipleCaption : "{count} files selected"
        }
        {...attributes}
      />
      <span className={styles["o-form__file-icon"]} aria-label={label}>
        <IconCloudUpload />
      </span>
      {children && (
        <span className={styles["o-form__file-text"]}>{children}</span>
      )}
    </label>
  );
};

export default File;
