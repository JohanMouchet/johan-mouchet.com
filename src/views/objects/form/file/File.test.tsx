import { render } from "@testing-library/react";
import { File } from "./File";

it("Renders <File /> without crashing", () => {
  expect(
    render(
      <File name="name" multipleCaption="Multiple Caption">
        Choose file
      </File>
    ).container.firstChild
  ).toMatchSnapshot();
});
