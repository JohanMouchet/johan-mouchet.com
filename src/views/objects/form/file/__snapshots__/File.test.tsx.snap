// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <File /> without crashing 1`] = `
<label
  class="o-form__file o-button"
>
  <input
    class="o-form__file-input u-is-visually-hidden"
    data-multiple-caption="Multiple Caption"
    name="name"
    type="file"
  />
  <span
    aria-label=""
    class="o-form__file-icon"
  >
    <svg />
  </span>
  <span
    class="o-form__file-text"
  >
    Choose file
  </span>
</label>
`;
